File: src/main/java/pangram/src/main/java/Pangrams.java
package pangram.src.main.java;

/**
 * Created by lusinabrian on 24/06/16.
 */
public class Pangrams {

    public static boolean isPangram(String pan){
        int i=0;
        for(char c : pan.toCharArray()) {
            int x = Character.toUpperCase(c);
            if (x >= 'A' && x <= 'Z') {
                i |= 1 << (x - 'A');
            }
        }
        return i == (i | ((1 << (1 + 'Z' - 'A')) - 1));
    }
}

File: ktstrings/src/main/kotlin/com/kotlinground/strings/pangram/Pangrams.kt
package com.kotlinground.strings.pangram

/**
 * @author lusinabrian on 08/06/17.
 */
object Pangrams {

    fun isPangram(sentence: String): Boolean {
        if (sentence.isNullOrEmpty())
            return false
        var i = 0
        sentence.toCharArray()
                .map { it.uppercaseChar().toInt() }
                .filter { it.toChar() in 'A'..'Z' }
                .forEach { i = i or (1 shl it.toChar() - 'A') }

        return i == i or (1 shl ('Z') + 1 - 'A') - 1
    }

    val alphaLength = 26
    fun isPangramX(input: String) = input.lowercase().replace(Regex("[^a-z]"), "").toSet().size == alphaLength

}

